src/Common/src/Interop/Unix/libc/Interop.getcwd.cs

```diff
@@ -61,12 +61,12 @@ private static unsafe string getcwd(byte* ptr, int bufferSize)
 
             // Otherwise, if it failed due to the buffer being too small, return null;
             // for anything else, throw.
-            int errno = Marshal.GetLastWin32Error();
-            if (errno == Interop.Errors.ERANGE)
+            ErrorInfo errorInfo = Interop.System.GetLastErrorInfo();
```

Comment:
:) I started down that path and then realized that this would be much simpler and easier to maintain.